### PR TITLE
Added button and menu upgrade docs

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Changes
- Moved shim stories under a common shims folder (instead of Button/Shims and Stack/Shims => Shims/Button and Shims/Stack)
- Added Button upgrade doc
- Added Menu upgrade doc

## Related Issue(s)

Update #22314

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/ce2398f2d96995250e8168676de60eacfd8e6a20'>ce2398f2d96995250e8168676de60eacfd8e6a20</a></sub>
<div id="bundleSizeComment9423"/>

File: apps/public-docsite-v9/src/Concepts/Upgrade/FromV8/Components/Button.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+| v8 Button variant    | v9 Button variant               |
+| -------------------- | ------------------------------- |
+| DefaultButton        | Button                          |
+| ActionButton         | Button appearance="transparent" |
+| CommandButton        | Button                          |
+| CommandBarButton     | Button                          |
+| CompoundButton       | CompoundButton                  |
+| ContextualMenuButton | Menu / MenuButton               |
+| PrimaryButton        | Button appearance="primary      |
```
Comment (khmakoto) [inline apps/public-docsite-v9/src/Concepts/Upgrade/FromV8/Components/Button.stories.mdx:23]: _nit:_ Missing "

```suggestion
| PrimaryButton        | Button appearance="primary"     |
```

File: apps/public-docsite-v9/src/Concepts/Upgrade/FromV8/Components/Button.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+| Button split         | SplitButton                     |
+| Button toggle        | ToggleButton                    |
+
+## Button Props Mapping
+
+Properties with Y in the Shim column are handled when using the button shims.
```
Comment (khmakoto) [inline apps/public-docsite-v9/src/Concepts/Upgrade/FromV8/Components/Button.stories.mdx:29]: What do you mean by "properties with Y in the Shim column"? I don't see any "Y"s there.

File: apps/public-docsite-v9/src/Concepts/Upgrade/FromV8/Components/Button.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+| disabled                 | disabled                              |
+| getClassNames            |                                       |
+| getSplitButtonClassNames | (use SplitButton)                     |
+| href                     | (use Link)                            |
+| iconProps                | icon (slot)                           |
+| disabled                 | disabled                              |
+| key                      | key                                   |
+| keyTipProps              |                                       |
+| menuAs                   | (use Menu and HTML as)                |
+| menuIconProps            | (use MenuButton menuIcon)             |
```
Comment (khmakoto) [inline apps/public-docsite-v9/src/Concepts/Upgrade/FromV8/Components/Button.stories.mdx:52]:
```suggestion
| menuIconProps            | (use MenuButton menuIcon slot)        |
```

File: apps/public-docsite-v9/src/Concepts/Upgrade/FromV8/Components/Button.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+| menuProps                | (use Menu and MenuButton)             |
+| menuTriggerKeyCode       |                                       |
+| onMenuClick              | HTML onClick                          |
+| onRenderAriaDescription  | (removed)                             |
+| onRenderChildren         | children                              |
+| onRenderDescription      | (use CompoundButton secondaryContent) |
```
Comment (khmakoto) [inline apps/public-docsite-v9/src/Concepts/Upgrade/FromV8/Components/Button.stories.mdx:58]:
```suggestion
| onRenderDescription      | (use CompoundButton secondaryContent slot) |
```

**Review by micahgodbolt — APPROVED** [review]

**Pull request merged**